File: tool/table/client/cardskillpassive.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let cardskillpassive_map = null;
export function cardskillpassive_map_init(config_obj:Object):void{
	cardskillpassive_map = config_obj["cardskillpassive_map"];
}


export class Cardskillpassive extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = cardskillpassive_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Cardskillpassive(key){
		if(Cardskillpassive.m_static_map.hasOwnProperty(key) == false){
			Cardskillpassive.m_static_map[key] = Cardskillpassive.create_Cardskillpassive(key);
		}
		return Cardskillpassive.m_static_map[key];	
	}
	public static create_Cardskillpassive(key){
		if(cardskillpassive_map.hasOwnProperty(key)){
			return new Cardskillpassive(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return cardskillpassive_map;
	}
}
}

File: app/config/cardskillpassive.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


cardskillpassive_map = {};
cardskillpassive_map[2001] = {"id":2001,"skilldata":[{"lv":1,"effect":"actor['atk']*=1.2","group":1,},{"lv":2,"effect":"actor['atk']*=1.3","group":1,},{"lv":3,"effect":"actor['atk']*=1.4","group":1,},],};
cardskillpassive_map[2002] = {"id":2002,"skilldata":[{"lv":1,"effect":"actor['spd']+=20","group":1,},{"lv":2,"effect":"actor['spd']+=30","group":1,},],};
cardskillpassive_map[2003] = {"id":2003,"skilldata":[{"lv":1,"effect":"actor['hpmax']*=1.2","group":1,},],};


class Cardskillpassive:
	def __init__(self, key):
		config = cardskillpassive_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Cardskillpassive(key):
		config = cardskillpassive_map.get(key);
		if not config:
			return
		return Cardskillpassive(key)

